### PR TITLE
UI/15399758 transactions

### PR DESCRIPTION
**Pull request by KatarzynaStefanowiczPagepro**
## Description

- [] TODO write description

**New stuff** ✨

* New stuff goes here

**Changes** 🏗

* Changes go here

**Deletions** ⚰️

* Deletions go here

## TODO

- [ ] Add TODOs

(Resolves | Contributes to) #31415

### CODE REVIEW

**Comment by KatarzynaStefanowiczPagepro** [comment]
https://tw.auroracreation.com/app/tasks/15399758

image

In the "Transactions" tab user can find a list of transactions. This is only a UI component - list and details when an item is expanded.

File: src/components/shared/Extensions/Accordion/hooks.ts [hunk 1]
```diff
@@ -1,7 +1,7 @@
 import { useState } from 'react';
 
 export const useAccordion = () => {
-  const [openIndex, setOpenIndex] = useState<number | undefined>(0);
+  const [openIndex, setOpenIndex] = useState<number | undefined>(-1);
```
Comment (krawczykkatarzyna) [inline src/components/shared/Extensions/Accordion/hooks.ts:4]: why this changes?

File: src/components/common/Extensions/UserHub/partials/TransactionsTab/types.ts [hunk 2]
```diff
@@ -0,0 +1,25 @@
+export interface TransactionsContent {
+  key: string;
+  title: string;
+  notificationInfo?: string;
+  status?: 'passed' | 'failed';
```
Comment (krawczykkatarzyna) [inline src/components/common/Extensions/UserHub/partials/TransactionsTab/types.ts:5]: please create union type and use it here

File: src/components/common/Extensions/UserHub/partials/TransactionsTab/types.ts [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+  isCurrentAction?: boolean;
+  isPending?: boolean;
+}
+
+export interface TransactionsItemProps {
+  key: string;
+  title: string;
+  description: string;
+  date: string;
+  status?: 'passed' | 'failed';
```
Comment (krawczykkatarzyna) [inline src/components/common/Extensions/UserHub/partials/TransactionsTab/types.ts:15]: the same

File: src/components/common/Extensions/UserHub/partials/TransactionsTab/partials/TransactionsItem/TransactionsItem.tsx [hunk 4]
```diff
@@ -0,0 +1,129 @@
...
+  description,
+  date,
+  status = 'passed',
+  content,
+  isOpen,
+  onClick,
+}) => {
+  const { formatMessage } = useIntl();
+
+  const transactionItemContent = (
```
Comment (krawczykkatarzyna) [inline src/components/common/Extensions/UserHub/partials/TransactionsTab/partials/TransactionsItem/TransactionsItem.tsx:24]: please move it to separate file

File: src/components/common/Extensions/UserHub/partials/TransactionsTab/partials/TransactionsItem/TransactionsItem.tsx [hunk 5]
```diff
@@ -0,0 +1,129 @@
...
+import clsx from 'clsx';
+import { useIntl } from 'react-intl';
+import { AnimatePresence, motion } from 'framer-motion';
+import Icon from '~shared/Icon/Icon';
+import NotificationBanner from '~common/Extensions/NotificationBanner/NotificationBanner';
+import styles from './TransactionsItem.module.css';
+import { TransactionsItemProps } from '../../types';
+import { accordionAnimation } from '~constants/accordionAnimation';
+
+export const displayName = 'common.Extensions.UserHub.partials.TransactionsItem';
```
Comment (krawczykkatarzyna) [inline src/components/common/Extensions/UserHub/partials/TransactionsTab/partials/TransactionsItem/TransactionsItem.tsx:11]: 'common/Extensions/UserHub/partials/TransactionsTab/partials/TransactionsItem'

File: src/components/common/Extensions/UserHub/partials/TransactionsTab/partials/TransactionsItem/TransactionsItem.tsx [hunk 6]
```diff
@@ -0,0 +1,129 @@
...
+                        <Icon
+                          name={item.status === 'passed' ? 'check-circle' : 'warning-circle'}
+                          appearance={{ size: 'tiny' }}
+                        />
+                      </div>
+                    )}
+                    {item.isPending && (
+                      <Icon
+                        name="spinner-gap"
+                        className="ml-[0.59375rem] w-[13px] h-[13px] animate-spin text-blue-400"
```
Comment (krawczykkatarzyna) [inline src/components/common/Extensions/UserHub/partials/TransactionsTab/partials/TransactionsItem/TransactionsItem.tsx:101]: rem

File: src/components/shared/Extensions/Accordion/hooks.ts (hunk 1 above)
Comment (KatarzynaStefanowiczPagepro) [reply src/components/shared/Extensions/Accordion/hooks.ts:4]: Is there any requirement that the accordion has to be opened by default?
I use this functionality in Transactions tabs and their default state is no one item opened.

File: src/components/common/Extensions/UserHub/partials/TransactionsTab/partials/TransactionsItem/TransactionsItem.tsx (hunk 6 above)
Comment (KatarzynaStefanowiczPagepro) [reply src/components/common/Extensions/UserHub/partials/TransactionsTab/partials/TransactionsItem/TransactionsItem.tsx:101]: fixed

File: src/components/common/Extensions/UserHub/partials/TransactionsTab/types.ts (hunk 3 above)
Comment (KatarzynaStefanowiczPagepro) [reply src/components/common/Extensions/UserHub/partials/TransactionsTab/types.ts]: fixed

File: src/components/common/Extensions/UserHub/partials/TransactionsTab/partials/TransactionsItem/TransactionsItem.tsx (hunk 4 above)
Comment (KatarzynaStefanowiczPagepro) [reply src/components/common/Extensions/UserHub/partials/TransactionsTab/partials/TransactionsItem/TransactionsItem.tsx]: fixed

**Review by krawczykkatarzyna — APPROVED** [review]
